WordPress/Classes/Services/ReaderCardService.swift

```diff
@@ -2,39 +2,54 @@ import Foundation
 
 class ReaderCardService {
     private let service: ReaderPostServiceRemote
+
     private let coreDataStack: CoreDataStack
+
     private let followedInterestsService: ReaderFollowedInterestsService
+
     private lazy var syncContext: NSManagedObjectContext = {
         return coreDataStack.newDerivedContext()
     }()
 
-    init(service: ReaderPostServiceRemote = ReaderPostServiceRemote(wordPressComRestApi: WordPressComMockrestApi()),
+    /// An string used to retrieve the next page
+    private var pageHandle: String?
+
+    /// Used only internally to order the cards
+    private var pageNumber = 1
+
+    init(service: ReaderPostServiceRemote = ReaderPostServiceRemote.withDefaultApi(),
          coreDataStack: CoreDataStack = ContextManager.shared,
          followedInterestsService: ReaderFollowedInterestsService? = nil) {
         self.service = service
         self.coreDataStack = coreDataStack
         self.followedInterestsService = followedInterestsService ?? ReaderTopicService(managedObjectContext: coreDataStack.mainContext)
     }
 
-    func fetch(page: Int = 1, success: @escaping (Int, Bool) -> Void, failure: @escaping (Error?) -> Void) {
+    func fetch(firstPage: Bool, success: @escaping (Int, Bool) -> Void, failure: @escaping (Error?) -> Void) {
         followedInterestsService.fetchFollowedInterestsLocally { [unowned self] topics in
             guard let interests = topics, !interests.isEmpty else {
                 failure(Errors.noInterests)
                 return
             }
 
             let slugs = interests.map { $0.slug }
-            self.service.fetchCards(for: slugs,
-                               success: { [weak self] cards in
+            let pageHandle: String? = firstPage ? nil : self.pageHandle
```

Comment:
Got stuck on this for a hot minute. Can you please change firstPage to isFirstPage? I think it would help

Reply:
Done in fe5e1d4a51b241599d1e78a706cec9ad05c4072c